ssc/cmod_battery.cpp

```diff
@@ -1944,15 +1985,57 @@ class cm_battery : public compute_module
                 n_rec_single_year,
                 dt_hour_gen);
 
+            // Setup custom forecasts
+            std::vector<ssc_number_t> p_pv_ac_forecast;
+            int batt_forecast_choice = as_integer("batt_dispatch_wf_forecast_choice");
+            if (is_assigned("batt_pv_ac_forecast")) {
+                p_pv_ac_forecast = as_vector_ssc_number_t("batt_pv_ac_forecast");
+                // Annual simulation is enforced above
+                if (p_pv_ac_forecast.size() < dt_hour_gen * 8760 && batt_forecast_choice == dispatch_t::WEATHER_FORECAST_CHOICE::WF_CUSTOM) {
+                    throw exec_error("pvsamv1", "batt_pv_clipping_forecast forecast length is " + std::to_string(p_pv_ac_forecast.size()) + " when custom weather file forecast is selected. Change batt_dispatch_wf_forecast_choice or provide a forecast of at least length " + std::to_string(dt_hour_gen * 8760));
+                }
+            }
+            else {
+                p_pv_ac_forecast = power_input_lifetime;
+            }
 
+            std::vector<ssc_number_t> p_load_forecast_in;
+            std::vector<ssc_number_t> p_load_forecast_full;
+            p_load_forecast_full.reserve(n_rec_lifetime);
+            if (is_assigned("batt_load_ac_forecast"))
+            {
+                p_load_forecast_in = as_vector_ssc_number_t("batt_load_ac_forecast");
+                size_t nload = p_load_forecast_in.size();
+                if (nload == 1) {
+                    // Length 1 is "empty" to UI lk
+                    p_load_forecast_in.clear();
+                }
+            }
+            if (p_load_forecast_in.size() > 0) {
```

Comment:
shouldn't this be `p_load_forecast_in.size() < weather file rec * analysis period`? a full life time forecast also has `p_load_forecast_in.size()` > 0

Reply:
The main load functions don't allow for lifetime loads. This is responding to the clear() on line 2011.

The length of load is far better enforced in pvsamv1 than it is in cmod_battery. I'll copy some of those functions over.

Reply:
ok thanks